Source/SwiftLintBuiltInRules/Rules/Lint/PrivateSwiftUIStatePropertyRule.swift

```diff
@@ -0,0 +1,52 @@
+import SwiftSyntax
+
+/// Rule to require that any state properties in SwiftUI be declared as private.
+/// State properties should only be accessible from inside the View's body, or from methods called by it
+struct PrivateSwiftUIStatePropertyRule: SwiftSyntaxRule, OptInRule, ConfigurationProviderRule {
+    var configuration = SeverityConfiguration<Self>(.warning)
+
+    static let description = RuleDescription(
+        identifier: "private_swiftui_state",
+        name: "Private SwiftUI @State Properties",
+        description: "SwiftUI's state properties should be private",
+        kind: .lint,
+        nonTriggeringExamples: PrivateSwiftUIStatePropertyRuleExamples.nonTriggeringExamples,
+        triggeringExamples: PrivateSwiftUIStatePropertyRuleExamples.triggeringExamples
+    )
+
+    func makeVisitor(file: SwiftLintFile) -> ViolationsSyntaxVisitor {
+        Visitor(viewMode: .sourceAccurate)
+    }
+}
+
+private extension PrivateSwiftUIStatePropertyRule {
+    final class Visitor: ViolationsSyntaxVisitor {
```

Comment:
I wonder if the rule should also check that the declaring type conforms to `View`. It doesn't seem too uncommon that there's an attribute called `State` which isn't the one SwiftUI defines.

Reply:
Good question, I believe I had played around with that at one point. Let me see if I can revive that code.